march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator_controller.py

```diff
@@ -0,0 +1,310 @@
+import os
+
+from numpy_ringbuffer import RingBuffer
+import rospy
+from trajectory_msgs.msg import JointTrajectory
+from urdf_parser_py import urdf
+
+from .model.modifiable_setpoint import ModifiableSetpoint
+from .model.modifiable_subgait import ModifiableSubgait
+
+
+class GaitGeneratorController(object):
+
+    def __init__(self, view):
+        self.view = view
```

Comment:
I think I would turn this dependency around. It seems weird to me that the controller needs a view in order to operate, and it makes it harder to test this class. Right now it is also responsible for connecting the buttons, which should not be something a controller should be aware of. Instead you want to create a controller and a view and connect all necessary buttons in the view to the controller.